src/components/LogView/index.js

```diff
@@ -130,32 +199,59 @@ class LogView extends React.Component<Props, State> {
     });
   }
 
-  genList = (index) => {
-    const line = this.props.filterData.filteredLines[index];
-    const lineNumber = line.lineNumber;
-    return (
-      <FullLogLine
-        lineRefCallback={this.lineRefCallback}
-        key={lineNumber}
-        found={lineNumber === this.props.findResults[this.props.searchFindIdx]}
-        bookmarked={this.findBookmark(this.props.bookmarks, lineNumber) !== -1}
-        highlight={this.props.highlights.highlightLines.includes(line)}
-        wrap={this.props.wrap}
-        line={line}
-        toggleBookmark={this.props.toggleBookmark}
-        colorMap={this.props.colorMap}
-        searchTerm={this.props.searchTerm}
-        highlightText={this.props.highlights.highlightText}
-        caseSensitive={this.props.caseSensitive}
-        updateSelectStartIndex={this.updateSelectStartIndex}
-        updateSelectEndIndex={this.updateSelectEndIndex}
-        handleDoubleClick={this.handleDoubleClick}
-      />
-    );
+  // :param ranges: list of 2d tuples
+  // The firt number in tuple means start line index, that will be expanded
+  // the second is end index
+  handleExpand = (ranges: Array<[number, number]>) => {
+    // TODO collapse sequential ranges into one
+    this.setState((state) => ({
+      expandedRanges: state.expandedRanges.concat(ranges),
+    }))
+  }
+
+  genList = (index: number) => {
+    const item = this.state.lines[index]
+    // disjoint union refinement
+    if (item.kind === 'SkipLine') {
+      // IF SkipLine
+      const skipLine: SkipLine = item
+      return (
+        <ExpandableLogLine
```

Comment:
These components don't extend all the way horizontally but they should:

![image](https://user-images.githubusercontent.com/29384396/51342121-03d5bd80-1a61-11e9-9507-a68488179768.png)